pkg/virt-operator/resource/apply/rbac.go

```diff
@@ -4,183 +4,250 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"k8s.io/apimachinery/pkg/api/equality"
+
+	"k8s.io/client-go/tools/cache"
+
+	"kubevirt.io/kubevirt/pkg/controller"
+
 	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"kubevirt.io/client-go/log"
 	"kubevirt.io/kubevirt/pkg/virt-operator/resource/generate/rbac"
 )
 
+type RoleType int
+
+const (
+	TypeRole               RoleType = iota
+	TypeClusterRole        RoleType = iota
+	TypeRoleBinding        RoleType = iota
+	TypeClusterRoleBinding RoleType = iota
+)
+
 func (r *Reconciler) createOrUpdateClusterRole(cr *rbacv1.ClusterRole, imageTag string, imageRegistry string, id string) error {
+	return r.createOrUpdate(cr, imageTag, imageRegistry, id, TypeClusterRole, false)
+}
+
+func (r *Reconciler) createOrUpdateClusterRoleBinding(crb *rbacv1.ClusterRoleBinding, imageTag string, imageRegistry string, id string) error {
+	return r.createOrUpdate(crb, imageTag, imageRegistry, id, TypeClusterRoleBinding, false)
+}
+
+func (r *Reconciler) createOrUpdateRole(role *rbacv1.Role, imageTag string, imageRegistry string, id string) error {
+	return r.createOrUpdate(role, imageTag, imageRegistry, id, TypeRole, true)
+}
 
-	var err error
-	rbac := r.clientset.RbacV1()
+func (r *Reconciler) createOrUpdateRoleBinding(rb *rbacv1.RoleBinding, imageTag string, imageRegistry string, id string) error {
+	return r.createOrUpdate(rb, imageTag, imageRegistry, id, TypeRoleBinding, true)
+}
 
-	var cachedCr *rbacv1.ClusterRole
+func (r *Reconciler) createOrUpdate(role interface{},
+	imageTag, imageRegistry, id string,
+	roleType RoleType,
+	avoidIfServiceAccount bool) (err error) {
```

Comment:
was this suppose to be rolebinding? serviceaccount is the name of another resource

Reply:
I get your confusion, the variable name is not ideal.
What I meant to express here is that some resources (e.g. role / role binding but not for cluster role / cluster role binding) do not need to be updated if a certain condition is satisfied (`!r.stores.ServiceMonitorEnabled && (roleMeta.Name == rbac.MONITOR_SERVICEACCOUNT_NAME`).
This condition was in the original functions for relevant resources. You'll the following condition within the function:
```
if avoidIfServiceAccount && !r.stores.ServiceMonitorEnabled && (roleMeta.Name == rbac.MONITOR_SERVICEACCOUNT_NAME) {
	return nil
}
```

Maybe more informative name like `avoidUpdateIfIsServiceMonitor` or something like that? What do you think?